readthedocs/core/resolver.py

```diff
@@ -255,29 +252,10 @@ def _fix_filename(self, project, filename):
         """
         Force filenames that might be HTML file paths into proper URL's.
 
-        This basically means stripping / and .html endings and then re-adding
-        them properly.
+        This basically means stripping /.
         """
-        # Bail out on non-html files
-        if '.' in filename and '.html' not in filename:
-            return filename
         filename = filename.lstrip('/')
-        filename = re.sub(r'(^|/)index(?:.html)?$', '\\1', filename)
-        if filename:
-            if filename.endswith('/') or filename.endswith('.html'):
-                path = filename
-            elif project.documentation_type == 'sphinx_singlehtml':
-                path = 'index.html#document-' + filename
-            elif project.documentation_type in ['sphinx_htmldir', 'mkdocs']:
-                path = filename + '/'
-            elif '#' in filename:
-                # do nothing if the filename contains URL fragments
-                path = filename
-            else:
-                path = filename + '.html'
-        else:
-            path = ''
-        return path
```

Comment:
I do feel like we introduced this for a reason, but perhaps we've removed that path from the codebase or something. I agree that the less we do with guessing URL's for things, the better, but I could still see this breaking in a random place.

Reply:
So, what I was thinking, this is used to reverse the url results from search, so all urls are present in the same format as the docs are build, rather than always including `.html`/`index.html`

Reply:
That's the only place where I can think it could break something, but when indexing from search we save the whole file, with the extension, no need to reverse anything from there. The other places we really don't use this at all.

Reply:
I'm still a little worried that this might break things, but I don't have a good way of testing it beyond just deploying it. @agjohnson do you have any memory of why we added this?